static/js/tools/map/fetcher/european_countries.ts

```diff
@@ -18,17 +18,21 @@
  * Fetch european countries.
  */
 
+import { useEffect, useState } from "react";
+
 import { EUROPE_NAMED_TYPED_PLACE } from "../../../shared/constants";
 import { NamedPlace } from "../../../shared/types";
 import { getEnclosedPlacesPromise } from "../../../utils/place_utils";
 
-export function fetchEuropeanCountries(
-  setEuropeanCountries: (data: Array<NamedPlace>) => void
-): void {
-  getEnclosedPlacesPromise(EUROPE_NAMED_TYPED_PLACE.dcid, "Country").then(
-    (resp: Array<NamedPlace>) => {
-      setEuropeanCountries(resp);
-      console.log("european countries loaded");
-    }
-  );
+export function useFetchEuropeanCountries(): Array<NamedPlace> {
```

Comment:
Does this need to be a useEffect? since it doesn't look like this is watching for any changes?

Reply:
In line 30, it is an async call, so here needs to "useState" to catch the response from the promise. 

Then needs useEffect() to work together here IIUC. It only fetches once (with [] dependency).

Reply:
ok, I see what you mean